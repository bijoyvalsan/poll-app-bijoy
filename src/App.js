import { Route, Switch } from "react-router-dom";
import {
  Questions,
} from "./pages";
import { Header } from './components';

import { routes } from "./config/routeConfig";

import { StyledContainer, StyledMain } from './App.styled';

const App = () => (
  <StyledContainer>
    <Header />
    <StyledMain id="main">
      <Switch>
        <Route exact path={routes.QUESTIONS} component={Questions} />
      </Switch>
    </StyledMain>
  </StyledContainer>
);

export default App;

