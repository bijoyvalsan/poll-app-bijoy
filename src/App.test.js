import React from "react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";

import App from "./App";
import configureStore from "./state/store";

const store = configureStore({});

describe("App Component", () => {
  it("should render", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
