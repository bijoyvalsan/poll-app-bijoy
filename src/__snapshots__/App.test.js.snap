// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render 1`] = `
<DocumentFragment>
  <div
    class="sc-dIvrsQ czLRA-d"
  >
    <header
      class="sc-bdnxRM giWmnp"
    >
      <h1>
        Poll App
      </h1>
    </header>
    <main
      class="sc-hHEiqL kDa-dOl"
      id="main"
    >
      <div
        class="sc-bqGGPW bvGpDM"
      >
        <div
          class="sc-ksluID eHYyPg"
        >
          <h2
            class="sc-hBMUJo igLrVS"
            data-testid="page-container-title"
          >
            Add Question
          </h2>
        </div>
        <div
          class="sc-iemWCZ jDoqWm"
        >
          <p
            class="sc-fnVZcZ idQmOh"
          >
            New Question
          </p>
          <div
            class="sc-fFSPTT fktQCA"
          >
            <input
              aria-label="Add Question here"
              data-testid="txt-question"
              id="txt-question"
              placeholder="Add Question here"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
          <p
            class="sc-fnVZcZ idQmOh"
          >
            Choices
          </p>
          <div
            class="sc-fFSPTT fktQCA"
          >
            <input
              aria-label="Choices seperated by comma"
              data-testid="txt-question-choice"
              id="txt-question-choice"
              placeholder="Choices seperated by comma"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
          <button
            class="sc-iCoGMd jtHMFE"
            data-testid="add-question-btn"
            disabled=""
          >
            Add Question
          </button>
        </div>
      </div>
      <div
        class="sc-bqGGPW bvGpDM"
      >
        <div
          class="sc-ksluID eHYyPg"
        >
          <h2
            class="sc-hBMUJo igLrVS"
            data-testid="page-container-title"
          >
            Questions
          </h2>
        </div>
        <div
          class="sc-bkbkJK iEoYTE"
        >
          <div
            class="sc-gtsrHT iCUPGs"
          >
            <div
              class="sc-dlnjwi sc-hKFxyN jkEfMo bGfFQW"
              width="60%"
            />
            <div
              class="sc-dlnjwi sc-hKFxyN hEEJwU eVeVZB"
              width="90%"
            />
            <div
              class="sc-dlnjwi sc-hKFxyN hEEJwU eVeVZB"
              width="90%"
            />
          </div>
          <div
            class="sc-gtsrHT iCUPGs"
          >
            <div
              class="sc-dlnjwi sc-hKFxyN jkEfMo bGfFQW"
              width="60%"
            />
            <div
              class="sc-dlnjwi sc-hKFxyN hEEJwU eVeVZB"
              width="90%"
            />
            <div
              class="sc-dlnjwi sc-hKFxyN hEEJwU eVeVZB"
              width="90%"
            />
          </div>
          <div
            class="sc-gtsrHT iCUPGs"
          >
            <div
              class="sc-dlnjwi sc-hKFxyN jkEfMo bGfFQW"
              width="60%"
            />
            <div
              class="sc-dlnjwi sc-hKFxyN hEEJwU eVeVZB"
              width="90%"
            />
            <div
              class="sc-dlnjwi sc-hKFxyN hEEJwU eVeVZB"
              width="90%"
            />
          </div>
          <div
            class="sc-gtsrHT iCUPGs"
          >
            <div
              class="sc-dlnjwi sc-hKFxyN jkEfMo bGfFQW"
              width="60%"
            />
            <div
              class="sc-dlnjwi sc-hKFxyN hEEJwU eVeVZB"
              width="90%"
            />
            <div
              class="sc-dlnjwi sc-hKFxyN hEEJwU eVeVZB"
              width="90%"
            />
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
